### PR TITLE
replaceBond and generic atoms and bonds in adjustQueryProperties()

### PR DESCRIPTION
**Pull request by greglandrum**
- adds `replaceBond()` functionality to `RWMol`
- adds converting atoms and/or bonds into generic queries to `adjustQueryProperties()`

Basic idea from Roger.

### CODE REVIEW

**Comment by bp-kelley** [comment]
If you are comfortable breaking the API, I'm happy with this PR.

**Comment by greglandrum** [comment]
The API is being broken anyway. :-)

**Comment by bp-kelley** [comment]
Yay!  Forward motion.

File: Code/GraphMol/MolOps.h [hunk 1]
```diff
@@ -243,13 +243,14 @@ void mergeQueryHs(RWMol &mol, bool mergeUnmappedOnly = false);
 
 typedef enum {
   ADJUST_IGNORENONE = 0x0,
-  ADJUST_IGNORECHAINATOMS = 0x1,
-  ADJUST_IGNORERINGATOMS = 0x4,
+  ADJUST_IGNORECHAINS = 0x1,
```
Comment (greglandrum) [inline Code/GraphMol/MolOps.h]: This piece of API does break, but it was only partially present in the previous release anyway. There are very few pieces of the rest of the code affected

**Comment by greglandrum** [comment]
Ignoring mapped atoms is already there: https://github.com/greglandrum/rdkit/blob/796834bf8c8761a5e7daa07031c6b62ba704b477/Code/GraphMol/MolOps.h#L250
You likely got confused by the diff view; that happens to me all the time.

**Pull request merged**